Make final adjustments before beta

davewiththenicehat opened the pull request:
Updates for this branch:

Done, consider adding a scrollbar to the settings window when there is more content than can fit on the page

Done, font size in center windows change immediately, but font size in left and right windows require a restart

Done, no way to change font size of labels

the copy says I need to restart after I change my background color, but actually I don't, it changes immediately
I think this works in linux, but not in windows?

Done, when I change my background color, it prints some copy that I don't understand. It's a numbered list of the six windows.

Done, Shorten change colors command

Done, Option to disable scroll bar on communication windows and main console.

Done, Change save location for client settings save to settings folder.

Make debug window close itself on `debug off`. Give it a scroll bar when needed.

https://github.com/TheEternalCitizens/mudlet-integration/issues/48

davewiththenicehat committed:
Map Resize Message

TEC Map will ask the user to walk to redraw the map after changing GUI.
Feature added because exits do not move on scale when map box is resized.

davewiththenicehat committed:
Settings Window & Update AntiSpam

Shortened settings window.
Added feature, if settings window has fewer rows available than text rows in the console a scroll bar appears. It disappears when the opposit occurs.
Minor disabled once and hour update check. Changed the text displayed when an update is not needed. ln3623

davewiththenicehat committed:
Scroll Bar Checker

Added function to check if a miniconsole needs a scroll bar.
Adds the scroll bar or removes is as needed.
Added this to room characters. If the list is longer than the window a scroll bar will apear. Once it shortens it will disappear.

davewiththenicehat committed:
FilterText Trigger hecho & Multiple Changes

Separated list of commands to change colors from main change colors help command & function.
Tried and failed to remove need to restart after changing text window backgrounds.
Notified user restart is required and that mudlet autosaves this change.
Fonts in all windows update with a font change. Not just main and com windows. user is notifed they need to save changes for a font size change.
user is notified they need to save on main text color change.
Created quick tutorial on moving color name chooser window.
Trigger for filtering html codes out of all lines and displays all main messages to screen now user hecho rather than hinsert.
hecho was sapose to allow for text background color changes via functions. I am uncertain why it doesn't work.
Also while playing I noticed the "You are no longer busy." in game notice would sometimes repeat itself.
In the past switching away from inserting text has had positive changes.

davewiththenicehat committed:
Changed settings save folder

Moved tec Auto Update function to start data and variables folder.
Used  tecUpdate.versionsRespitory to create string tecSettingsFolder. It is used to set settings folder location.
Changed load and save folders for all functions including guiFrame.

davewiththenicehat committed:
TabsText, label refresh

tecConfigThemes() created to refresh theme variables
Geyser.Container:objectQuery() created, collects data on a geyser object and ALL it's children objects. To a table.
tecRedrawLabels() created, redraws project geyser labels
Created tecClientSetTabFontSize() it sets the font size for projects geyser tabs and refreshes the labels to draw the changes.
scripts ln 4663 corrected.

davewiththenicehat committed:
Added ability disable scrollbars

Aliases
added commands to enable or disable projects scroll bars.
Variable tecSettings.showScrollBars if true scrollbars will be shown.
How to change GUIframe in two locations. 1649 - 1653 and 2118 - 2121
Added function fuzzyBoolean(bool) made by demonnic. Will check for many incorrect data types and return the correct boolean. "yes" "true" or number 0 return true "no" "false" number 1 is false.
Had to change setMainWindowAutoWrap() lns 3027 - 3064
Created function tecSetScrollBars(mode) it enables or disables scroll bars. Handles repositioning on setting.
Added help for new feature in tecDisplayHelp()lns 3492 - 3501
Created script space `Post Initilization` intended for code that needs to be processed when the project has completed startup.

davewiththenicehat committed:
Color Fix, Removed remaining Borders, minor debug upgrade

Aliases
Moved a echo for tecclient background color into function where is should be.
Scripts
Removed borders that still remain. Many lines throughout the project. 1492, 3275, 5034, 5042, 5057, 5060, 5083, 50865, 5093 - 5094, 5101
Added option to enable scroll bar for debug window. ln 2682
debug window closes on `debug off` ln 2689
Made color words with _ in the name compatible with the project's color charger. ln 4038, 4055, 4074
Added `don't forget to save your changes` Notice in the color change functions.
If color function was missing a description at the top it has been added
Color change functions that affect room checks now auto refresh when run. ln4170, 4185

davewiththenicehat commented:
testing webhook

johnnymo87 reviewed (approved):
Really nice work, @davewiththenicehat :clap:

davewiththenicehat committed:
Version Update

Updated version numbers to beta.
triggers ln 409 added a trigger to hopefully fix issue #14
